### PR TITLE
f/wpc-320-create-an-author-in-the-article-metadata

### PR DESCRIPTION
**Pull request by HanaAyman**
I added a button that will appear when searching for an author that doesn't exist in the article metadata pannel and it will create the author and make it appear automatically so the publisher can then select it.

### CODE REVIEW

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 1]
```diff
@@ -380,6 +421,16 @@ export function ArticleMetadataPanel({
                     setAuthorsFilter('')
                   }}
                   block
+                  renderExtraFooter={() =>
+                    authorsFilter &&
+                    data?.authors.nodes.length === 0 && (
```
Comment (bishoymelek) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:426]: can we have it like `!data?.authors.nodes.length && (` , what do you think?

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 2]
```diff
@@ -380,6 +421,16 @@ export function ArticleMetadataPanel({
...
+                      <div style={{float: 'right', margin: '20px'}}>
```
Comment (bishoymelek) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:427]: I guess using `float` has side effects and you will need to clearfix the parent. you can use flex for example what do you think?

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 3]
```diff
@@ -380,6 +421,16 @@ export function ArticleMetadataPanel({
...
+                        <Button onClick={() => handleCreateAuthor()} appearance={'primary'}>
```
Comment (bishoymelek) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:428]: `appearance='primary'`

**Review by bishoymelek — CHANGES_REQUESTED** [review]
Amazing work @HanaAyman, just very simple comments

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 4]
```diff
@@ -137,6 +143,23 @@ export function ArticleMetadataPanel({
     }
   }, [data?.authors, authors, socialMediaAuthors])
 
+  const [createAuthor] = useCreateAuthorMutation({
+    refetchQueries: [getOperationNameFromDocument(AuthorListDocument)]
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:147]: Why did you add this? What is the purpose of that?

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 5]
```diff
@@ -137,6 +143,23 @@ export function ArticleMetadataPanel({
...
+  })
+  const [authorName, setAuthorName] = useState('')
+  const [authorSlug, setAuthorSlug] = useState('')
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:150]: i don't think you need a state for the slug. You can create the slug in `handleCreateAuthor()`

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 6]
```diff
@@ -211,7 +234,11 @@ export function ArticleMetadataPanel({
                     value: author.id,
                     label: author.name
                   }))}
-                  onSearch={searchKeyword => setAuthorsFilter(searchKeyword)}
+                  onSearch={searchKeyword => {
+                    setAuthorsFilter(searchKeyword)
+                    setAuthorName(searchKeyword)
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:239]: Do you even need a state for `authorName` you could also take the value from `authorsFilter`.

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 7]
```diff
@@ -222,6 +249,16 @@ export function ArticleMetadataPanel({
                     setAuthorsFilter('')
                   }}
                   block
+                  renderExtraFooter={() =>
+                    authorsFilter &&
+                    data?.authors.nodes.length === 0 && (
+                      <div style={{float: 'right', margin: '20px'}}>
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:255]: I would not use float right.

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 8]
```diff
@@ -222,6 +249,16 @@ export function ArticleMetadataPanel({
...
+                        <Button onClick={() => handleCreateAuthor()} appearance={'primary'}>
+                          {t('authors.panels.createAuthor')}
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:257]: Please make the label of the button more descriptive. "Create Auhtor Profile for %Name%"

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 9]
```diff
@@ -380,6 +421,16 @@ export function ArticleMetadataPanel({
                     setAuthorsFilter('')
                   }}
                   block
+                  renderExtraFooter={() =>
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:424]: you seem to repeat yourself here. It might be a good idea to create a new component callled AuthorCheckPicker and move the logic there.

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Looks very good just some inputs.

File: packages/editor/src/client/panel/articleMetadataPanel.tsx (hunk 2 above)
Comment (HanaAyman) [reply packages/editor/src/client/panel/articleMetadataPanel.tsx:427]: It's not needed anymore since I changed the button's name. 👍

File: packages/editor/src/client/panel/articleMetadataPanel.tsx (hunk 4 above)
Comment (HanaAyman) [reply packages/editor/src/client/panel/articleMetadataPanel.tsx:147]: I use it to make the new author I just created appear automatically in the list after clicking the button so the user can then select it without needing to refresh. Is there a better solution ?

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 10]
```diff
@@ -222,6 +245,16 @@ export function ArticleMetadataPanel({
...
+                    authorsFilter &&
+                    !data?.authors.nodes.length && (
+                      <div style={{margin: '20px'}}>
+                        <Button onClick={() => handleCreateAuthor()} appearance="primary">
+                          {t('articles.panels.createAuthorProfile', {name: authorsFilter})}
```
Comment (penk85) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:253]: It's a nice touch to add the user name to the comment, but as a personal preference I find it distracting that the button is dynamically changing size. Maybe the comment can be in a label and the button should have a static text "Create"?

**Review by penk85 — CHANGES_REQUESTED** [review]

File: packages/editor/src/client/panel/articleMetadataPanel.tsx (hunk 4 above)
Comment (roosnic1) [reply packages/editor/src/client/panel/articleMetadataPanel.tsx:147]: ahh. This is awesome. Good job

File: packages/editor/src/client/panel/articleMetadataPanel.tsx (hunk 10 above)
Comment (roosnic1) [reply packages/editor/src/client/panel/articleMetadataPanel.tsx:253]: I think it's good if the name is in the button. Otherwise the users might think it will open the `AuthorEditPanel` and that's not what happens.

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Awesome work. Please remove all the commented code and then we can approve. 🎉

**Review by bishoymelek — APPROVED** [review]
:tada: Great work @HanaAyman

**Review by penk85 — APPROVED** [review]

**Review by roosnic1 — APPROVED** [review]
🥳
